power_grid_model_c/power_grid_model_cpp/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+# C++ API library
+add_library(power_grid_model_cpp INTERFACE)
+
+target_include_directories(power_grid_model_cpp INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+
+set(PGM_PUBLIC_HEADERS
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/basics.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/buffer.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/dataset.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/handle.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/meta_data.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/model.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/options.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp/serialization.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/power_grid_model_cpp.hpp"
+)
```

Comment:
because you use `install(DIRECTORY ...)` below, you don't use this variable. Please remove (and also double-check whether the same is true for the C API) OR (even better) if we already require CMake version `3.19` or higher, then you can add sources to the `add_library(INTERFACE ...)` cfr. https://cmake.org/cmake/help/latest/command/add_library.html#interface-with-sources

Reply:
You are correct. I can just remove `set(PGM_PUBLIC_HEADERS ...)` for this wrapper. In the case of the C API, I believe things are a bit different, because this is set, but no `install(DIRECTORY ...)`, and it makes sense to do it explicitly there as there is documentation for the C API, whereas, at least for now, this wrapper is for internal use only.

On the other hand, can we just jump to CMake version 3.19? I see that we currently support `cmake_minimum_required (VERSION 3.9)`, so we are "very behind", and I wonder if that change may be a problem from users that build PGM themselfs? Shouldn't we bring this up for discussion if needed?

Reply:
The top-level `CMakeLists.txt` has `cmake_minimum_required (VERSION 3.23)`

Reply:
oh wait, on the `export` side it's indeed `3.9`

Reply:
Is it then safe to change to `3.23` since it is used in some places already as that version?

Reply:
let's discuss tomorrow with tony. I'm not sure. The exported target can have less strict requirements than the imported target.

Reply:
Alright. Makes sense.

Reply:
@TonyXiang8787 do you have an opinion on changing the cmake version from `3.9` to `3.23` on the `export` side?

Reply:
I agree. We don't deliver our C-API to package manger. So the user needs to build it from source. To build them from source you need `3.23` anyway.